src/scripts/docker_migration_apt.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+# Set environment variables to avoid interactive prompts
+export DEBIAN_FRONTEND=noninteractive
+DOCKER_DOWNLOAD_ORIGINS="Docker:\${distro_codename}"
+UNATTENDED_UPGRADES_FILE="/etc/apt/apt.conf.d/50unattended-upgrades"
+LOG_FILE="/usr/src/dappnode/logs/upgrade_015.log"
+
+log() {
+  echo "$(date '+%Y-%m-%d %H:%M:%S') $1" | tee -a ${LOG_FILE}
+}
+
+log "Starting docker install migration from pkg to apt"
+
+# Only update docker if unattended upgrades is enabled,
+# otherwise the update might crash due to updating docker from docker.
+if [ ! -f "${UNATTENDED_UPGRADES_FILE}" ]; then
+  log "WARNING: Unattended upgrades is not enabled, skipping upgrade"
+  exit 0
+fi
+
+# Check if docker is installed via apt
+# The docker.list file is created by the docker installation script
+if [ -f /etc/apt/sources.list.d/docker.list ]; then
+  log "Docker is already installed via apt, skipping upgrade"
+  # TODO: check if docker is in unattended upgrades
+  exit 0
+fi
+
+# Get docker version
+DOCKER_VERSION=$(docker version --format '{{.Server.Version}}')
+
+# Check the OS
+if [ -f /etc/os-release ]; then
+  source /etc/os-release
+  OS=$NAME
+  VERSION=$VERSION_ID
+elif type lsb_release >/dev/null 2>&1; then
+  OS=$(lsb_release -si)
+  VERSION=$(lsb_release -sr)
+elif [ -f /etc/lsb-release ]; then
+  source /etc/lsb-release
+  OS=$DISTRIB_ID
+  VERSION=$DISTRIB_RELEASE
+elif [ -f /etc/debian_version ]; then
+  OS=Debian
+  VERSION=$(cat /etc/debian_version)
+else
+  OS=$(uname -s)
+  VERSION=$(uname -r)
+fi
+
```

Comment:
Maybe you should stop installation if none of the above branches is `true`?

Reply:
already exiting in next step